ddl/column.go

```diff
@@ -693,17 +705,360 @@ func (w *worker) doModifyColumn(
 		}
 	})
 
-	if newAutoRandBits > 0 {
-		if err := checkAndApplyNewAutoRandomBits(job, t, tblInfo, newCol, oldName, newAutoRandBits); err != nil {
+	if updatedAutoRandomBits > 0 {
+		if err := checkAndApplyNewAutoRandomBits(job, t, tblInfo, newCol, oldColName, updatedAutoRandomBits); err != nil {
```

Comment:
We need to ensure the column type is unchanged when modifying `auto_random` bits.

Reply:
"auto_random" requires a primary key, but it does not support modifying the column type with primary key.